### PR TITLE
Eswords

### PR DESCRIPTION
**Pull request by SweptWasTaken**
Added Blue, Green, Red and Purple eswords. They are currently all separate entities but ideally I want a single esword that has it's color changed by a multi-tool. (And of course a rainbow variation).

Some issues with this pr i'd like feedback/help on.

-**Getting pointlight to correctly display on inhand sprites.**-

I want the eswords to glow correctly on the inhand sprites. In this pr they glow like **this.**

Look cool but it's obvious the light is emanating from the player and not the esword itself. The solution would be to add an `offset` to the eswords pointlight like so.

Great! But it's borked when you try looking left and right. Additionally, dual wielding eswords causes the light to appear in the wrong place due to the offset.
Content Client_2020-06-03_17-36-35
There's also this strange thing where the pointlight rotates as you click.

-**Disabling the "melee" function of the esword when it's sheathed.**-

-**A way to adjust pointlight radius/energy/offset with sprite state changes or set to a timer**-

This would just be useful for stuff that flickers like a flare/fire/esword beam.

-**I will add a way to put two eswords together and make a double esword**-

-**I'm tired bro**-

### CODE REVIEW

File: Content.Server/GameObjects/Components/Weapon/Melee/EswordComponent.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            if (Activated)
+            {
+                TurnOff();
+            }
+            else
+            {
+                TurnOn(user);
+            }
+
+            return true;
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Weapon/Melee/EswordComponent.cs:54]: Why did you make this return something?

File: Content.Server/GameObjects/Components/Weapon/Melee/EswordComponent.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+
+        public void Examine(FormattedMessage message)
+        {
+            var loc = IoCManager.Resolve<ILocalizationManager>();
+
+            if (Activated)
+            {
+                message.AddMarkup(loc.GetString("The light is currently [color=darkgreen]on[/color]."));
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Weapon/Melee/EswordComponent.cs:114]: You can just use the static `Loc` class here, I think.

**Comment by ComicIronic** [comment]
> There's also this strange thing where the pointlight rotates as you click.

This is because clicking "turns" you in the direction of the click, and point light sources by default turn with their parent.

Fixing the offset seems quite complicated. You'd somehow have to have the server know what offset to use for the hand the esword is being held in.

**Comment by SweptWasTaken** [comment]
Closed unless someone else wants to work on it.

**Comment by PaulRitter** [comment]
maybe we should add a derelict tag (like /vg/) for prs that got closed but still have salvagable content? (this)

**Pull request closed without merging**